### PR TITLE
OCPBUGS-14907: Fix operator backed catalog page when copied CSVs disabled

### PR DESCRIPTION
**Pull request by vikram-raj**
Fixes: https://issues.redhat.com/browse/OCPBUGS-14907

Descriptions: 
Show catalog items in operator backed even if copied to CSVs is disabled.

### CODE REVIEW

**Review by TheRealJon — COMMENTED** [review]
There are two main cases we need to cover when listing CSVs and copied CSVs are disabaled:

1. **Listing in a namespaced context:** We are currently in a namespace other than 'openshift', 'openshift-operators', or 'All Namespaces'.
2. **We are listing in a global context:** we are in one of the three excluded namespaces from (1). 



### **Namespaced context** 
List CSVs from the current namespace and copied CSVs from the 'openshift' namespace

### **Global Context**
 a. **All Namespaces:** List CSVs from all namespaces, but filter out copied CSVs. (This is probably already covered since this is not a new case).
 b. **'openshift-operators' namespace:** List all operators in this namespace only and they are all available in all namespaces (they have copies in the 'openshift' namespace).
 c. **'openshift' namespace:** List all operators in this namespace only. Copied CSVs are available in all namespaces, otherwise, they are only available in the current namespace.
 
@tlwu2013 Could you help confirm this logic? Mainly with (b). Are all CSVs in the 'openshift-operators' namespace considered global or can namespaced operators be installed their as well?

**Comment by tlwu2013** [comment]
https://github.com/openshift/console/pull/12932#pullrequestreview-1495809640

> @tlwu2013 Could you help confirm this logic? Mainly with (b). Are all CSVs in the 'openshift-operators' namespace considered global or can namespaced operators be installed their as well?


> Are all CSVs in the ‘openshift-operators’ namespace considered global

yes, all Operators in `openshift-operators` namespace are global.
(but global Operators can exist somewhere else, like Serverless does)

> or can namespaced operators be installed their as well?

no, they cannot. 

--
if users tried to install an Operator as namespaced scope in `openshift-operators` namespace, they got blocked by this:
> **Namespace does not support installation mode**
> The openshift-operators Namespace is reserved for global Operators that watch all Namespaces. To install an Operator in a single Namespace, select a different Namespace where the operand should run.

**Comment by TheRealJon** [comment]
Thanks @tlwu2013!

@vikram-raj As Tony mentioned, we also need to handle the case where an operator is installed globally, but not in the default `openshift-operators` namespace. He used Serverless as an example. By default, that operator is installed globally in the `openshift-serverless` namespace. This means the source CSV will be in that namespace, while a copy will be made in the 'openshift' namespace. Therefore, when listing CSVs in the 'openshift-serverless' namespace, the source CSV should be listed and not the copied CSV from 'openshift'. I believe this case is handled in the admin perspective, we just need to make sure it's consistent in the dev perspective too.

**Comment by vikram-raj** [comment]
@TheRealJon I updated the can you PTAL again. Thanks.

**Review by TheRealJon — CHANGES_REQUESTED** [review]

File: frontend/packages/operator-lifecycle-manager/src/utils/useClusterServiceVersions.tsx [hunk 1]
```diff
@@ -143,17 +144,34 @@ const useClusterServiceVersions: ExtensionHook<CatalogItem[]> = ({
...
+      namespace: GLOBAL_COPIED_CSV_NAMESPACE,
+      prop: referenceForModel(ClusterServiceVersionModel),
+      optional: window.SERVER_FLAGS.copiedCSVsDisabled,
+    }),
+    [],
+  );
+  const resourceSelector =
+    window.SERVER_FLAGS.copiedCSVsDisabled && !GLOBAL_OPERATOR_NAMESPACES.includes(namespace)
+      ? globalResourceSelector
+      : namespacesResourceSelector;
```
[the remark this answers is not in the document]
Comment (vikram-raj) [reply frontend/packages/operator-lifecycle-manager/src/utils/useClusterServiceVersions.tsx]: Updated it PTAL again. Thanks

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**